dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/runner/WorkerTaskExecuteRunnable.java

```diff
@@ -216,15 +217,17 @@ protected void beforeExecute() {
         TaskExecutionCheckerUtils.downloadResourcesIfNeeded(storageOperate, taskExecutionContext, logger);
         logger.info("Resources:{} check success", taskExecutionContext.getResources());
 
+        TaskFilesTransferUtils.downloadUpstreamFiles(taskExecutionContext, storageOperate);
```

Comment:
I think it's not elegant enough to simply download zip file before executing and upload file after executing, cuz intermediate data maybe large-scale. Slice and buffer write will be better and also bring some benefits for prefetching for downstream task.

Reply:
Good idea, but I think this should be implemented in A in the resource center interface because the resource center can connect HDFS, S3, and OSS. . Maybe we can create a new issue to do this, a unified interface can be provided later. WDYT?

Reply:
Sound good, maybe we can design it together, some mechanism in distributed filesystem are worth referring to.

Reply:
I have created a new issue about that #12578